### PR TITLE
ECE340 Course Offering Page

### PR DESCRIPTION
**Pull request by egwhite3**

### CODE REVIEW

File: docs/Course Wiki/ECE Course Offerings/ECE340.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# ECE340
+
```
Comment (lnguy7) [inline docs/Course Wiki/ECE Course Offerings/ECE340.md:5]: https://github.com/hkn-alpha/wiki/pull/6#discussion_r1251461268
Add sth like this

File: docs/Course Wiki/ECE Course Offerings/ECE340.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+## Content Covered
+
+- Band diagrams
+- Fermi-Dirac/Maxwell-Boltzmann Statistics
+- Drift and Diffusion of Carriers
+- PN Junctions
+- MOSCAPs and MOSFETs
+- BJTs
+- Optoelectronics (Photodetectors, Solar Cells, LEDs, LASERs)
```
Comment (lnguy7) [inline docs/Course Wiki/ECE Course Offerings/ECE340.md]: Preferably format like this
https://github.com/hkn-alpha/wiki/pull/6#discussion_r1251461683

**Review by lnguy7 — CHANGES_REQUESTED** [review]
Looks good but Idk the content so Im looking more at the formatting Ill have Alex look at content lol

File: docs/Course Wiki/ECE Course Offerings/ECE340.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+## Prerequisites
+
+- [PHYS214](../PHYS%20Course%20Offerings/PHYS214.md)
+- [ECE329](ECE329.md) (Corequisite)
```
Comment (alex-m-zhang) [inline docs/Course Wiki/ECE Course Offerings/ECE340.md:27]: ECE210 is also an explicit prerequisite iirc.

File: docs/Course Wiki/ECE Course Offerings/ECE340.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+## When to Take it
+
+Taking this course in accordance with the stated prerequisites and co-requisites is especially important for ECE 340, due to its emphasis on conceptual understanding. Because of the overlap with physics concepts, it is best to take this as soon as possible after the prerequisites. This course is a gateway to many other courses in electrical engineering such as electronic circuits and IC fabrication, and in physics. It is beneficial for students considering careers in anything to do with physical and quantum electronics and circuits to take ECE 340 as early as possible.
```
Comment (alex-m-zhang) [inline docs/Course Wiki/ECE Course Offerings/ECE340.md]: Add how to take this course ASAP (just mention smth about taking 210 asap, then taking this concurrently with 329. PHYS 214 will probably be completed by the time you take 210, but if not, take 210 with 214).

File: docs/Course Wiki/ECE Course Offerings/ECE340.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+## Course Structure
+
+The course is fairly standard with a combination of 11 homeworks, 2 midterms, and a final comprising to total grade.
```
Comment (alex-m-zhang) [inline docs/Course Wiki/ECE Course Offerings/ECE340.md]: The course's **structure** is fairly standard...
(add the word structure)

Comment (alex-m-zhang) [inline docs/Course Wiki/ECE Course Offerings/ECE340.md]: You can delete "comprising to total grade.".

File: docs/Course Wiki/ECE Course Offerings/ECE340.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+## Course Structure
+
+The course is fairly standard with a combination of 11 homeworks, 2 midterms, and a final comprising to total grade. 
+
+Homework assignments are assigned each week and usually consist of three or four problems that may involve drawing band diagrams, derivations, calculations, conceptual explanations, or using an online semiconductor simulation applet. There will be multiple times where certain assumption must be made (temperature independence, neglecting a small concentration, etc) to make ceratin problems solable in a reasonable amount of time. This can be frustrating for students used to solving problems with the exact parameters given to them, but making and justifying these assumptions are part of the course.
```
Comment (alex-m-zhang) [inline docs/Course Wiki/ECE Course Offerings/ECE340.md]: "There will be multiple times where certain assumptions must be made" (add s after assumption)
Certain is spelled incorrectly

File: docs/Course Wiki/ECE Course Offerings/ECE340.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+Midterm exams for ECE340 are quite imfamous among the 300 level ECE courses. While slight logistical things can vary between semesters (like an equation sheet), in general the tests are 3 questions with several subpart to be solved in 1 hour without a calculator. They are usually even more of a test of time management than ECE210 exams. The strangest quirk about this course is the lack of access to past exams. The testing format also means that there is little similarity to homework problems. This structure makes understanding the concepts taught in ECE340 paramount to success.
```
Comment (alex-m-zhang) [inline docs/Course Wiki/ECE Course Offerings/ECE340.md]: several "subparts" to be solved in 1 hour (add s after subpart)
Can you add what the different logistical things surrounding the cheatsheet are? Would be good to know.

File: docs/Course Wiki/ECE Course Offerings/ECE340.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+Try not to get discouraged when recieving grades back, this is set up to be a tough course and your overall grade does not fully reflect your ability to work with semiconductors. 
+
+Partial credit on exams is your best friend. If a problem is too challenging, write down as many equations or notes that seem adjacent enough to the problem and the graders may reward it. 
+
+ALWAYS justify an assumption. It is required to complete a problem, but also a good check to make sure the assumption is valid.
+
+## Life After
+
+Not surprisingly, ECE340 is the prerequisite to many courses in the area of solid-state electronics, including ECE441 (Physics and Modeling of Semiconductor Devices), ECE444 (Theory and Fabrication of Integrated Circuits), and ECE482 (Digital IC Design). Taking [ECE342](ECE342.md) (Electronic Circuits) either before, concurrently, or after ECE340 provides a good overview of electronic circuits at both the device and circuit levels.
```
Comment (alex-m-zhang) [inline docs/Course Wiki/ECE Course Offerings/ECE340.md]: You can change these courses into links, even if the links don't exist yet.

File: docs/Course Wiki/ECE Course Offerings/ECE340.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+## Infamous Topics
```
Comment (alex-m-zhang) [inline docs/Course Wiki/ECE Course Offerings/ECE340.md:61]: For each one, also add how to combat it or something. Ex:
The whole class: Not to worry, the curve will save you.
PN Junctions: Keep track of xyz, do yxe, etc.

File: docs/Course Wiki/ECE Course Offerings/ECE340.md [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+- The whole class
+- PN Junctions
+- MOSFETs
+
+## Resources
+
+[nanohub.org](https://nanohub.org/) is run by those scumbags at Purdue, and unfortunatly it is one of the best semiconductor resources ever created. Not only does it have amazing introductory videos and lecutres, it also has a built in simulation tool to get a "hands on" understanding of semiconductors.
+
+
+THE TEXTBOOK IS YOUR FRIEND
```
Comment (alex-m-zhang) [inline docs/Course Wiki/ECE Course Offerings/ECE340.md]: Don't the lectures not follow the order of the textbook? Is there a mapping of what lecture to what textbook section or smth?
Also doesn't the textbook go way further in depth than required?

(What you wrote is fine, just add some more detail)

File: docs/Course Wiki/ECE Course Offerings/ECE340.md [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+
+## Prerequisites
+
+- [PHYS214](../PHYS%20Course%20Offerings/PHYS214.md)
+- [ECE329](ECE329.md) (Corequisite)
+
+Officially [PHYS214](../PHYS%20Course%20Offerings/PHYS214.md) is listed as a prerequisite, while [ECE329](ECE329.md) is listed as a pre-/co-requisite. ECE340 involves many physics concepts such as diffusion and particle distribution, so having completed [PHYS213](../PHYS%20Course%20Offerings/PHYS213.md) is almost as important as PHYS214. In fact, several PHYS213 lectures cover semiconductor physics at a basic level. The first part of ECE340 will involve various topics from PHYS214 such as the concept of discrete energy levels, and a brief discussion of Schrödinger's Equation and its consequences in semiconductors. Relevant topics from ECE329 include Maxwell's equations in material media and carrier mobility. They apply in ECE340 to topics such as p-n junctions and carrier flow, but should be covered early enough in the semester if taken concurrently with ECE 340. Overall, these courses are most helpful in developing an intuition for what is happening at the physics level inside electronic devices.
```
Comment (alex-m-zhang) [inline docs/Course Wiki/ECE Course Offerings/ECE340.md]: Paragraph break before talking about ECE329 is helpful.

**Review by alex-m-zhang — CHANGES_REQUESTED** [review]

File: docs/Course Wiki/ECE Course Offerings/ECE340.md (hunk 1 above)
Comment (alex-m-zhang) [reply docs/Course Wiki/ECE Course Offerings/ECE340.md:5]: For this, just add a sentence about how this is required for EE and a 1-of-6 for CE.

**Review by alex-m-zhang — CHANGES_REQUESTED** [review]

File: docs/Course Wiki/ECE Course Offerings/ECE340.md [hunk 12]
```diff
@@ -45,17 +56,17 @@ ALWAYS justify an assumption. It is required to complete a problem, but also a g
...
-Not surprisingly, ECE340 is the prerequisite to many courses in the area of solid-state electronics, including ECE441 (Physics and Modeling of Semiconductor Devices), ECE444 (Theory and Fabrication of Integrated Circuits), and ECE482 (Digital IC Design). Taking [ECE342](ECE342.md) (Electronic Circuits) either before, concurrently, or after ECE340 provides a good overview of electronic circuits at both the device and circuit levels.
+Not surprisingly, ECE340 is the prerequisite to many courses in the area of solid-state electronics, including [ECE441](ECE441.md) (Physics and Modeling of Semiconductor Devices), [ECE444](ECE444.md) (Theory and Fabrication of Integrated Circuits), and [ECE482](ECE482.md) (Digital IC Design). Taking [ECE342](ECE342.md) (Electronic Circuits) either before, concurrently, or after ECE340 provides a good overview of electronic circuits at both the device and circuit levels.
 
 ## Infamous Topics
 
-- The whole class
-- PN Junctions
-- MOSFETs
+- The whole class (But this is true for everyone, so the curve will save you.)
+- PN Junctions (Keep track of subscripts, the variable is WHAT is described and the subscript is (ususally) WHERE it is)
```
Comment (alex-m-zhang) [inline docs/Course Wiki/ECE Course Offerings/ECE340.md]: Usually is spelled wrong.

File: docs/Course Wiki/ECE Course Offerings/ECE340.md [hunk 13]
```diff
@@ -1,35 +1,46 @@
 # ECE340
+##Semiconductor Electronics.
+
+ECE 340 is the introductory semiconductors course that is required for EE and a 1-of-6 for CE.
```
Comment (alex-m-zhang) [inline docs/Course Wiki/ECE Course Offerings/ECE340.md]: ECE340 (COURSE NAME HERE) is a 3-credit-hour course that is specifically required for EEs as part of the Electrical Engineering Core and satisfies the 1-of-6 Electrical Engineering Foundations Course requirement for CEs. It is offered in the fall, spring, and summer.

File: docs/Course Wiki/ECE Course Offerings/ECE340.md [hunk 14]
```diff
@@ -45,17 +56,17 @@ ALWAYS justify an assumption. It is required to complete a problem, but also a g
 
 ## Life After
 
-Not surprisingly, ECE340 is the prerequisite to many courses in the area of solid-state electronics, including ECE441 (Physics and Modeling of Semiconductor Devices), ECE444 (Theory and Fabrication of Integrated Circuits), and ECE482 (Digital IC Design). Taking [ECE342](ECE342.md) (Electronic Circuits) either before, concurrently, or after ECE340 provides a good overview of electronic circuits at both the device and circuit levels.
+Not surprisingly, ECE340 is the prerequisite to many courses in the area of solid-state electronics, including [ECE441](ECE441.md) (Physics and Modeling of Semiconductor Devices), [ECE444](ECE444.md) (Theory and Fabrication of Integrated Circuits), and [ECE482](ECE482.md) (Digital IC Design). Taking [ECE342](ECE342.md) (Electronic Circuits) either before, concurrently, or after ECE340 provides a good overview of electronic circuits at both the device and circuit levels.
```
Comment (alex-m-zhang) [inline docs/Course Wiki/ECE Course Offerings/ECE340.md]: Format as:
[ECExxx - title of course here](ECExxx.md)

**Review by alex-m-zhang — CHANGES_REQUESTED** [review]
Almost done - just some last formatting stuff. Sorry.

**Review by lnguy7 — APPROVED** [review]
I have no prevailing issue with the formatting or anything for now so LGTM but idk abt content wise

**Review by alex-m-zhang — APPROVED** [review]
LGTM!

**Pull request merged**